### PR TITLE
doc: update version and release notes for v1.0.0

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1431)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,56 @@
 # Changelog
 
-## v0.10.x - TBD
+## v1.1.0 - TBD
```
Comment (devjgm) [inline CHANGELOG.md]: Would the next version be 1.0.1?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,56 @@
...
+
+* **Other Changes**
+
+* feat: add Managed Backup feature (#1358)
+* feat: sessions are automatically created and refreshed by
+  background threads as described in:
+      https://cloud.google.com/spanner/docs/sessions#best_practices
+* We made multiple improvements (detailed below) to the documentation and
+  code samples.
+
```
Comment (devjgm) [inline CHANGELOG.md]: Is this blank line needed?

File: CHANGELOG.md (hunk 1 above)
Comment (devjgm) [reply CHANGELOG.md]: Nevermind, my mistake.

Comment (coryan) [reply CHANGELOG.md]: No.  Next version is a minor release. We do patch releases only for well, patches.

**Comment by coryan** [comment]
Rebased and updated the API baseline.

File: CHANGELOG.md (hunk 2 above)
Comment (coryan) [reply CHANGELOG.md]: Nope, removed.

**Pull request merged**